src/ui/public/vislib/visualizations/_point_series_chart.js

```diff
@@ -1,91 +1,66 @@
 import d3 from 'd3';
 import _ from 'lodash';
-import VislibVisualizationsChartProvider from 'ui/vislib/visualizations/_chart';
-import VislibComponentsTooltipProvider from 'ui/vislib/components/tooltip';
+import ChartProvider from 'ui/vislib/visualizations/_chart';
+import TooltipProvider from 'ui/vislib/components/tooltip';
 import errors from 'ui/errors';
 
 export default function PointSeriesChartProvider(Private) {
 
-  const Chart = Private(VislibVisualizationsChartProvider);
-  const Tooltip = Private(VislibComponentsTooltipProvider);
+  const Chart = Private(ChartProvider);
+  const Tooltip = Private(TooltipProvider);
   const touchdownTmpl = _.template(require('ui/vislib/partials/touchdown.tmpl.html'));
 
   class PointSeriesChart extends Chart {
     constructor(handler, chartEl, chartData) {
       super(handler, chartEl, chartData);
     }
 
-    _stackMixedValues(stackCount) {
-      let currentStackOffsets = [0, 0];
-      let currentStackIndex = 0;
-
-      return function (d, y0, y) {
-        const firstStack = currentStackIndex % stackCount === 0;
-        const lastStack = ++currentStackIndex === stackCount;
+    getStackedCount() {
+      return _.reduce(this.handler.data.get('series'), function (sum, val) {
+        if (val.stacked || 1) return sum + 1;
+      }, 0);
+    };
 
-        if (firstStack) {
-          currentStackOffsets = [0, 0];
-        }
+    getStackedNum(data) {
+      let i = 0;
+      for (const seri of this.handler.data.get('series')) {
+        if (seri === data) return i;
+        if (seri.stacked || 1) i++;
+      }
+      return 0;
+    };
 
-        if (lastStack) currentStackIndex = 0;
+    getValueAxis() {
+      return _.find(this.handler.valueAxes, axis => {
+        return axis.id === this._attr.valueAxis;
+      }) || this.handler.valueAxes[0];
+    };
 
-        if (y >= 0) {
-          d.y0 = currentStackOffsets[1];
-          currentStackOffsets[1] += y;
-        } else {
-          d.y0 = currentStackOffsets[0];
-          currentStackOffsets[0] += y;
-        }
-      };
+    getCategoryAxis() {
```

Comment:
yup, thats how it works. however i like the  "if it doesnt the first axis is used"  ... its a fallback behaviour for the case when you have just one axis (would feel stupid you would need to provide axisId to every series even if there is just one axis)